lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -2,6 +2,42 @@
 
 module PolicyMachineStorageAdapter
   class ActiveRecord
+    class PolicyElementAssociation
+      def self.with_accessible_operation(associations, operation)
+        query = <<-SQL
+          operation_set_id IN (
+            WITH RECURSIVE accessible_operations AS (
+              (
+                SELECT
+                  child_id,
+                  parent_id,
+                  parent_id AS operation_set_id
+                FROM assignments
+                WHERE parent_id IN (#{associations.select(:operation_set_id).to_sql})
```

Comment:
is this faster than 
```
  WHERE parent_id IN (:operation_set_ids)
. . .
WHERE ops.unique_identifier = :operation
)
SQL

associations.where(query, operation_set_ids: associations.pluck(:operation_set_id), operation: operation)
```
which i think is more standard rails query

Reply:
As is, this implementation compresses that logic into a single query vs two queries (one for the `.pluck` and one when it gets resolved), but I haven't timed the difference.